Code/Framework/AzCore/Tests/StatisticalProfiler.cpp

```diff
@@ -5,847 +5,651 @@
+        ASSERT_TRUE(profiler.GetStatsManager().AddStatistic(statIdThread4, statNameThread4, "us"));
+        ASSERT_TRUE(profiler.GetStatsManager().AddStatistic(statIdThread4Loop, statNameThread4Loop, "us"));
 
         proxy->ActivateProfiler(ProfilerProxyGroup, true);
 
-        //Let's kickoff the threads to see how much contention affects the profiler's performance.
-        const int iter_count = 10;
-        AZStd::thread t1(AZStd::bind(&simple_thread1, iter_count));
-        AZStd::thread t2(AZStd::bind(&simple_thread2, iter_count));
-        AZStd::thread t3(AZStd::bind(&simple_thread3, iter_count));
+        AZStd::thread t1([&](){
+            RecordStatistics(MediumIterationCount, statIdThread1, statIdThread1Loop);
+        });
+        AZStd::thread t2([&](){
+            RecordStatistics(MediumIterationCount, statIdThread2, statIdThread2Loop);
+        });
+        AZStd::thread t3([&](){
+            RecordStatistics(MediumIterationCount, statIdThread3, statIdThread3Loop);
+        });
+        AZStd::thread t4([&](){
+            RecordStatistics(MediumIterationCount, statIdThread4, statIdThread4Loop);
+        });
         t1.join();
         t2.join();
         t3.join();
+        t4.join();
 
         ASSERT_TRUE(profiler.GetStatistic(statIdThread1) != nullptr);
         EXPECT_EQ(profiler.GetStatistic(statIdThread1)->GetNumSamples(), 1);
         ASSERT_TRUE(profiler.GetStatistic(statIdThread1Loop) != nullptr);
-        EXPECT_EQ(profiler.GetStatistic(statIdThread1Loop)->GetNumSamples(), iter_count);
+        EXPECT_EQ(profiler.GetStatistic(statIdThread1Loop)->GetNumSamples(), MediumIterationCount);
 
         ASSERT_TRUE(profiler.GetStatistic(statIdThread2) != nullptr);
         EXPECT_EQ(profiler.GetStatistic(statIdThread2)->GetNumSamples(), 1);
         ASSERT_TRUE(profiler.GetStatistic(statIdThread2Loop) != nullptr);
-        EXPECT_EQ(profiler.GetStatistic(statIdThread2Loop)->GetNumSamples(), iter_count);
+        EXPECT_EQ(profiler.GetStatistic(statIdThread2Loop)->GetNumSamples(), MediumIterationCount);
 
         ASSERT_TRUE(profiler.GetStatistic(statIdThread3) != nullptr);
         EXPECT_EQ(profiler.GetStatistic(statIdThread3)->GetNumSamples(), 1);
         ASSERT_TRUE(profiler.GetStatistic(statIdThread3Loop) != nullptr);
-        EXPECT_EQ(profiler.GetStatistic(statIdThread3Loop)->GetNumSamples(), iter_count);
+        EXPECT_EQ(profiler.GetStatistic(statIdThread3Loop)->GetNumSamples(), MediumIterationCount);
+
+        ASSERT_TRUE(profiler.GetStatistic(statIdThread4) != nullptr);
+        EXPECT_EQ(profiler.GetStatistic(statIdThread4)->GetNumSamples(), 1);
+        ASSERT_TRUE(profiler.GetStatistic(statIdThread4Loop) != nullptr);
+        EXPECT_EQ(profiler.GetStatistic(statIdThread4Loop)->GetNumSamples(), MediumIterationCount);
 
-        //Clean Up
         proxy->ActivateProfiler(ProfilerProxyGroup, false);
     }
+}//namespace UnitTest
```

Comment:
Would it be a lot of work to separate the testing code to a different file and/or lib from the benchmark?

Reply:
Splitting them into different files wouldn't be a problem, different libs would be a completely separate story though.